### PR TITLE
HAI-654 Create base Coder Ubuntu image

### PR DESCRIPTION
**Pull request by jamesconstable**
This PR does the initial setup for a Coder images repo, starting with a base Ubuntu image that includes some common ML prerequisites (see [ticket](https://harrison-ai.atlassian.net/jira/software/projects/HAI/boards/80?selectedIssue=HAI-654) for full requirements).

Please also treat the repo itself as part of this PR; I've tried to configure it to behave similarly to our other repos, but I don't have admin rights on any others so I can't compare their settings directly. @andychisholm, @dcarrion87 and @schattingh, I've added you all as admins, so feel free to go ahead and make any corrections needed on that front.

### CODE REVIEW

**Comment by jamesconstable** [comment]
Some additional loose ends on this PR:
 * The RapidFile toolkit is missing. Downloading it requires a PureStorage account which I'm not sure how to manage from within the `docker build` step, and its licence wouldn't allow us to download it once and then drop it into our open source repo. I looked around the other infra repos to see how it was done for the mgmt machines, but I'm guessing it must've been installed manually?
 * I've copied the Docker Hub upload step from `cobalt-docker-terraform`, but don't have the secrets to make it work.

On the question of how much `apt-utils` + `dialog` + running `unminimize` bloats image size, the difference is 2.35GB without vs 3.19GB with. I've omitted them from the current version, but let me know if you think we should include them.

**Comment by dcarrion87** [comment]
@jamesconstable 

> * The RapidFile toolkit is missing. Downloading it requires a PureStorage account which I'm not sure how to manage from within the `docker build` step, and its licence wouldn't allow us to download it once and then drop it into our open source repo. I looked around the other infra repos to see how it was done for the mgmt machines, but I'm guessing it must've been installed manually?

Good call. I didn't even think that dropping it in our open sourced container image is actually breaking the licence. Let's just leave it out and we'll provide an alternate way of users getting at it after the container comes up if they really need it.

**Review by andychisholm — APPROVED** [review]
Nice!

RE
> On the question of how much apt-utils + dialog + running unminimize bloats image size, the difference is 2.35GB without vs 3.19GB with. I've omitted them from the current version, but let me know if you think we should include them.

I'd say this is small enough that we throw @anaoum a bone here and do it.

File: base/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ENV PATH="${PATH}:/home/coder/.local/bin"
+
+ARG AWS_VERSION=2.7.18
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64-${AWS_VERSION}.zip" -o "/tmp/awscliv2.zip" && \
+    unzip -q /tmp/awscliv2.zip -d /tmp && \
+    /tmp/aws/install && \
+    rm -rf /tmp/aws /tmp/awscliv2.zip
+
+ARG KUBE_VERSION=1.22.12
+RUN curl -LO "https://dl.k8s.io/release/v${KUBE_VERSION}/bin/linux/amd64/kubectl" && \
```
Comment (dcarrion87) [inline base/Dockerfile:43]: This is going to be an interesting one how it goes with different kube versions users might be interacting with in their environments. Leave it for now though.

**Review by dcarrion87 — APPROVED** [review]
Let's see how it goes!

**Pull request merged**